plugins/c8y_configuration_plugin/src/download.rs

```diff
@@ -128,97 +145,235 @@ pub async fn handle_config_download_request_child_device(
+
+    // NOTE: this tedge needs to go
+    let path_to = PathBuf::from(format!(
+        "{FILE_TRANSFER_ROOT_PATH}/{}",
+        config_management.http_file_repository_relative_path()
+    ));
+    let temp_download_file_name = &format!(".tmp_{}", config_file_name);
+    let downloader = Downloader::new(temp_download_file_name, &None, FILE_TRANSFER_ROOT_PATH);
     downloader.download(&download_info).await?;
+    downloader.rename(&path_to).await?;
 
-    // Read the config file contents to be uploaded to the file transfer service
-    let config_content = std::fs::read_to_string(downloader.filename())?;
-    debug!("Downloaded file content: {config_content}");
-
-    // Generate a unique target URL path for the child device to download the config file update from the file trasnsfer service
-    let config_file_url_path =
-        format!("/tedge/file-transfer/{child_id}/config_update/{config_type}");
-
-    // Upload the config file to file transfer service
-    let config_file_url = format!("http://{local_http_host}{config_file_url_path}");
-    info!("Uploading downloaded config update to {config_file_url}");
-    let _response = Client::new()
-        .put(&config_file_url)
-        .body(config_content)
-        .send()
-        .await?;
-
-    debug!("Deleting the config file update temporary copy downloaded from Cumulocity");
-    downloader.cleanup().await?;
-
-    info!("Sending config update request to child device");
-    let child_req_topic = Topic::new_unchecked(
-        format!("tedge/{child_id}/commands/req/config_update/{config_type}").as_str(),
+    let msg = Message::new(
+        &config_management.child_topic_request(),
+        config_management.request_payload(local_http_host)?,
     );
-    let msg = Message::new(&child_req_topic, config_file_url_path);
     mqtt_client.published.send(msg).await?;
 
     Ok(())
 }
 
-pub async fn handle_config_update_executing_response_child_device(
-    message: Message,
-    mqtt_client: &mut Connection,
-) -> Result<(), anyhow::Error> {
-    let child_id = get_child_id_from_child_response_topic(&message.topic.name);
-    let c8y_child_topic = Topic::new_unchecked(format!("c8y/s/us/{child_id}").as_str());
-    let executing_status_payload = DownloadConfigFileStatusMessage::status_executing()?;
-    mqtt_client
-        .published
-        .publish(Message::new(&c8y_child_topic, executing_status_payload))
-        .await?;
+#[derive(Serialize, Deserialize)]
+pub struct ChildDevicePayload {
+    pub url: String,
+    pub path: String,
+    pub config_type: Option<String>,
+}
 
-    Ok(())
+#[derive(Serialize, Deserialize)]
+pub struct ChildDeviceP {
+    pub status: OperationStatus,
+    path: String,
+    #[serde(rename = "type")]
+    pub config_type: String,
+    reason: Option<String>,
+}
+
+pub enum ChildResponse {
+    Executing(Message),
+    Failed(Message),
+    Successful(Option<Message>),
+    Updated(Message),
+}
+
+pub fn config_snapshot_child_device_message(
```

Comment:
That name looks really misleading as this module is about `download` for handling `config_update` requests and it has a method named `config_snapshot`. Should this be moved to the `upload` module?